bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -785,12 +786,35 @@ private void handlePushActivity(@Nullable String payload) {
         }
 
         String search = key.registeredUserId + "#" + key.entryId;
-        connection.getActivities(10, pushActivity.timestamp).stream().filter(activity -> search.equals(activity.id))
-                .findFirst()
-                .ifPresent(currentActivity -> currentActivity.getSourceDeviceIds().stream()
-                        .map(sourceDeviceId -> findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber))
-                        .filter(Objects::nonNull).forEach(echoHandler -> Objects.requireNonNull(echoHandler)
-                                .handlePushActivity(currentActivity)));
+
+        /*
+         * DEPRECATED
+         * connection.getActivities(10, pushActivity.timestamp).stream().filter(activity -> search.equals(activity.id))
+         * .findFirst()
+         * .ifPresent(currentActivity -> currentActivity.getSourceDeviceIds().stream()
+         * .map(sourceDeviceId -> findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber))
+         * .filter(Objects::nonNull).forEach(echoHandler -> Objects.requireNonNull(echoHandler)
+         * .handlePushActivity(currentActivity)));
+         */
+
+        Long timestamp = pushActivity.timestamp;
+        if (timestamp != null) {
+            Long startTimestamp = timestamp - 30000;
+            Long endTimestamp = timestamp + 30000;
+            List<CustomerHistoryRecord> customerHistoryRecords = connection.getActivities(startTimestamp, endTimestamp);
+            for (CustomerHistoryRecord customerHistoryRecord : customerHistoryRecords) {
+                if (customerHistoryRecord != null) {
+                    String recordKey = customerHistoryRecord.recordKey;
+                    if (recordKey != null && search.equals(recordKey)) {
```

Comment:
```suggestion
                    if (search.equals(recordKey)) {
```

No need to check for null: `search` is not null and .equals is always null-safe (for the second object)

Reply:
the check for null is not because of the following equals-method, it is because of the use of recordKey inside the if statement.